src/EditorFeatures/VisualBasic/LineCommit/CommitFormatter.vb

```diff
@@ -87,6 +90,20 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
             End Using
         End Sub
 
+        Private Function AbortForDiagnostics(document As Document, textSpanToFormat As TextSpan, cancellationToken As CancellationToken) As Boolean
+            Const UnterminatedStringId = "BC30648"
+
+            Dim tree = document.GetSyntaxTreeAsync(cancellationToken).WaitAndGetResult(cancellationToken)
+
+            ' If we have any unterminated strings that overlap what we're trying to format, then
+            ' bail out.  It's quite likely the unterminated string will cause a bunch of code to
+            ' swap between real code and string literals, and committing will just cause problems.
+            Dim diagnostics = tree.GetDiagnostics(cancellationToken).Where(
```

Comment:
I could.  But this was nicer to debug, since i could take a look at the diagnostics and determine which i needed, and what i would do with them.  The 'Any' approach saves a single allocation.  So i stuck with the more debuggable code (to me).